src/pages/index.js

```diff
@@ -229,5 +229,7 @@ function handleSubmit(request, modalInstance) {
   request()
     .then(() => modalInstance.close())
     .catch(console.error)
-    .finally(() => modalInstance.renderLoading(false));
+    .finally(() => {
+        formValidators[modalInstance.formName].resetValidation();
```

Comment:
* `deleteConfirmationManager` doesn't have a validator, so you need to protect your code from errors.


It's done like this `formValidators[modalInstance.formName]?.resetValidation();` . You should use `?` before the **dot**

```js
function handleSubmit(request, modalInstance) {
  modalInstance.renderLoading(true);
  request()
    .then(() => {
      modalInstance.close();
      formValidators[modalInstance.formName]?.resetValidation();  // here
    })
    .catch(console.error)
    .finally(() => {
      modalInstance.renderLoading(false);
    });
}


```

Reply:
I accomplished this by refactoring the ModalConfirm class. I moved the logic for handling the delete out of the makeNewCard function and into the deleteConfirmationManager declaration. 
By doing so, I could:
- rope ModalConfirm's handling into the handleSubmit function, drying these up quite a bit
- clean up the ModalConfirm class to be way more readable/simple